File: java/BasicMath.java
package java;

public class BasicMath {
    public static int add(int a, int b) {
        return a + b;
    }
    
    public static int subtract(int a, int b) {
        return a - b;
    }
    
    public static int multiply(int a, int b) {
        return a * b;
    }
    
    public static double divide(int a, int b) {
        return (double) a / b;
    }
    
    public static int power(int base, int exponent) {
        if (exponent < 0) {
            throw new IllegalArgumentException("Negative exponent not supported");
        }
        
        int result = 1;
        for (int i = 0; i < exponent; i++) {
            result *= base;
        }
        return result;
    }

    public static int factorial(int n) {
        if (n < 0) {
            throw new IllegalArgumentException("Negative numbers not supported!");
        }
        int result = 1;
        for (int i = 1; i <= n; i++) {
            result *= i;
        }
        return result;
    }
}

File: kotlin/BasicMath.kt
package kotlin

object BasicMath {
    fun add(a: Int, b: Int): Int = a + b
    
    fun subtract(a: Int, b: Int): Int = a - b
    
    fun multiply(a: Int, b: Int): Int = a * b
    
    fun divide(a: Int, b: Int): Double {
        require(b != 0) { "Division by zero" }
        return a.toDouble() / b
    }
    
    fun power(base: Int, exponent: Int): Int {
        require(exponent >= 0) { "Negative exponent not supported" }
        
        var result = 1
        for (i in 0 until exponent) {
            result *= base
        }
        return result
    }
    
    fun factorial(n: Int): Long {
        require(n >= 0) { "Factorial not defined for negative numbers" }
        
        var result: Long = 1
        for (i in 2..n) {
            result *= i
        }
        return result
    }
}